### PR TITLE
Add real-time H3 index reader

### PR DESCRIPTION
**Pull request by yupeng9**
Add the H3 index support to real-time segments.

Updated the RealTimeQuickStart example to utilize the H3 index. For example

image

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * A H3 index reader for the real-time H3 index values on the fly.
+ * <p>This class is thread-safe for single writer multiple readers.
+ */
+public class RealtimeH3IndexReader implements GeoSpatialIndexCreator, H3IndexReader {
+  private final H3Core _h3Core;
+  private final Map<Long, ThreadSafeMutableRoaringBitmap> _h3IndexMap = new ConcurrentHashMap<>();
+  private final H3IndexResolution _resolution;
+  private int _lowestResolution;
+
+  public RealtimeH3IndexReader(List<Integer> resolutions)
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java:26]: why multiple constructors? lets stick with H3IndexConfig?

File: pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+
+
+/**
+ * A H3 index reader for the real-time H3 index values on the fly.
+ * <p>This class is thread-safe for single writer multiple readers.
+ */
+public class RealtimeH3IndexReader implements GeoSpatialIndexCreator, H3IndexReader {
+  private final H3Core _h3Core;
+  private final Map<Long, ThreadSafeMutableRoaringBitmap> _h3IndexMap = new ConcurrentHashMap<>();
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java:22]: may be start with something bigger depending on the resolution to avoid resizes

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/column/ColumnIndexContainer.java [hunk 3]
```diff
@@ -51,7 +51,7 @@
   /**
    * Returns the H3 index for the column, or {@code null} if it does not exist.
    */
-  H3IndexReader getH3Index();
+  ImmutableH3IndexReader getH3Index();
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/index/column/ColumnIndexContainer.java:54]: why is this change needed?

**Review by kishoreg — APPROVED** [review]
LGTM. @Jackie-Jiang can you take a look at it

File: pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java (hunk 1 above)
Comment (yupeng9) [reply pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java:26]: done

File: pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java (hunk 2 above)
Comment (yupeng9) [reply pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/RealtimeH3IndexReader.java:22]: there is a tradeoff of memory usage?

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/column/ColumnIndexContainer.java (hunk 3 above)
Comment (yupeng9) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/index/column/ColumnIndexContainer.java:54]: H3IndexReader is changed to interface

**Review by kishoreg — APPROVED** [review]

**Comment by kishoreg** [comment]
@yupeng9 missing header

**Comment by yupeng9** [comment]
@kishoreg Updated.
